Shell_Sort_and_Quick_Sort.cpp

```diff
@@ -0,0 +1,408 @@
+void QuickSortRecursive( vector<collegeType> & tempS, int start, int end, int choice );
+void WriteTimeRecords( string fileName, string fname, float readTime, float runTime, float writeTime );
+
+int main(void) { 
+	string fileName;
+	float rTime = 0.0;
+	vector<collegeType> cSet; 				// set of output records
+	readRecords( cSet, fileName, rTime ); 	// call: read records from a file
+	displayAll(cSet); 						// call: output all on screen
+	shellSort( cSet, fileName, rTime ); 	// call: shell sort
+	quickSort( cSet, fileName, rTime ); 	// call: quick sort
+	system("pause");
+	return 0;
+} // end main
+//*********************************************************************************/
+
+bool readRecords(vector<collegeType> &cSet, string & fileName, float & rTime ) {   // read records from a file
+	clock_t timer;
+	cout << "����ɦW(601~605)�G";
+	cin >> fileName;
+	fstream inFile; 						// input file handle
+	string temp = fileName;
+	fileName = "input" + fileName + ".txt"; // input file name is fixed
+ 	timer = clock();						// start the timer
+	 
+	string coName, depName, stLevel;
+ 	char rBuf[255]; 								// input buffer for one line
+ 	inFile.open(fileName.c_str(), fstream::in); 	// open file to read
+ 	if (inFile.is_open()) 							// unable to open file
+ 		cout << endl << "### " << fileName << " does not exist! ###" << endl;
+ 	else {
+ 		while (inFile.getline(rBuf, 255, '\n')) { 	// get each input record line by line
+ 			string temp; 							// transform the c-string of input record
+ 			collegeType oneC; 						// output record of one college
+ 			int fNo = 0; 							// number of fields read so far
+ 			int pre = 0, pos = 0;					// locate one field of input record
+ 			temp.assign(rBuf); 						// copy string as a c-string
+ 			pos = temp.find_first_of('\t', pre); 	// locate the first field by '\t' (tab)
+ 			while (pos != string::npos) { 			// continue if it is not the end of record
+  				switch (++fNo) {
+  					case 1: oneC.cname = temp.substr(pre, pos - pre);
+ 						break; // get college name
+ 					case 2: oneC.dname = temp.substr(pre, pos - pre);
+ 						break; // get department name
+ 					case 3: oneC.level = temp.substr(pre, pos - pre);
+ 						break; // get level
+ 					default: break;
+ 				} //end switch
+ 				
+ 				pre = ++pos; 							// move to the next field
+ 				pos = temp.find_first_of('\t', pre); 	// locate the next delimiter
+ 			} //end inner-while
+ 			
+ 			pos = temp.find_last_of('\t');
+ 			temp = temp.substr(pos+1); 			// get total
+ 			oneC.total = atoi(temp.c_str()); 	// transform total into integer type
+ 			cSet.push_back(oneC); 				// save one record into the vector (array)
+ 		} //end outer-while
+ 	}	// else
+ 	
+ 	timer = clock();						// start the timer
+ 	timer = clock() - timer; 	// get the elapse time
+ 	cout << endl << "Ū�ɮɶ��G" << timer << " clocks (";
+ 	cout << ( (float)timer ) / CLOCKS_PER_SEC << " seconds)." << endl;
+ 	rTime = ( (float)timer ) / CLOCKS_PER_SEC ;
```

Comment:
Duplicate logic
( (float)timer ) / CLOCKS_PER_SEC is computed twice.
```suggestion
        rTime = ( (float)timer ) / CLOCKS_PER_SEC ;
        cout << endl << "讀檔時間：" << timer << " clocks (";
 	cout << rTime << " seconds)." << endl;
```